Error When Creating SBOM for Image Specified with Digest

#### What happened:

An error occurred when creating an SBOM for a container image specified using a digest.

#### What you expected to happen:

I expected no error to happen when creating an SBOM for a container image specified using a digest.

#### How to reproduce it (as minimally and precisely as possible):

```bash
bom generate --image alpine@sha256:8914eb54f968791faf6a8638949e480fef81e697984fba772b3976835194c6d4
```

Error:

```
panic: interface conversion: name.Reference is name.Digest, not name.Tag

goroutine 1 [running]:
sigs.k8s.io/bom/pkg/spdx.refInfoFromIndex(0xc000002140)
	sigs.k8s.io/bom/pkg/spdx/implementation.go:253 +0xed3
sigs.k8s.io/bom/pkg/spdx.getImageReferences({0xc0000de870, 0x4e})
	sigs.k8s.io/bom/pkg/spdx/implementation.go:211 +0x25d
sigs.k8s.io/bom/pkg/spdx.(*spdxDefaultImplementation).PullImagesToArchive(0x0?, {0xc0000de870?, 0x15755c3?}, {0xc0000de960, 0x45})
	sigs.k8s.io/bom/pkg/spdx/implementation.go:359 +0x4c
sigs.k8s.io/bom/pkg/spdx.(*spdxDefaultImplementation).ImageRefToPackage(0xc0000ca000?, {0xc0000de870, 0x4e}, 0x1e?)
	sigs.k8s.io/bom/pkg/spdx/implementation.go:698 +0x172
sigs.k8s.io/bom/pkg/spdx.(*SPDX).ImageRefToPackage(...)
	sigs.k8s.io/bom/pkg/spdx/spdx.go:253
sigs.k8s.io/bom/pkg/spdx.(*defaultDocBuilderImpl).GenerateDoc(0x0?, 0x1988850, 0xc0000c10e0)
	sigs.k8s.io/bom/pkg/spdx/builder.go:234 +0xb6a
sigs.k8s.io/bom/pkg/spdx.(*DocBuilder).Generate(0xc0000edb50?, 0x1?)
	sigs.k8s.io/bom/pkg/spdx/builder.go:96 +0xb5
sigs.k8s.io/bom/cmd/bom/cmd.generateBOM(0xc0000c0fc0)
	sigs.k8s.io/bom/cmd/bom/cmd/generate.go:341 +0x3db
sigs.k8s.io/bom/cmd/bom/cmd.AddGenerate.func1(0xc0001cc000?, {0xc0000c7760?, 0x2?, 0x2?})
	sigs.k8s.io/bom/cmd/bom/cmd/generate.go:146 +0xe8
github.com/spf13/cobra.(*Command).execute(0xc0001cc000, {0xc0000c7740, 0x2, 0x2})
	github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0x19805e0)
	github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.6.1/command.go:968
sigs.k8s.io/bom/cmd/bom/cmd.Execute()
	sigs.k8s.io/bom/cmd/bom/cmd/root.go:71 +0x25
main.main()
	./main.go:24 +0x17
```

#### Anything else we need to know?:

Is this the intended behavior? If only tags are supported on purpose, then please close this issue. Thank you for this great tool!

#### Environment:

OS: M1 Mac w/ macOS 13.0.1
Go: 1.19.2

--- a/pkg/spdx/implementation.go
+++ b/pkg/spdx/implementation.go
@@ -248,9 +248,9 @@ func refInfoFromIndex(descr *remote.Descriptor) (refinfo *ImageReferenceInfo, er
 	logrus.Infof("Reference image index points to %d manifests", len(indexManifest.Manifests))
 	refinfo.MediaType = string(indexManifest.MediaType)
 
-	// Add all the child images describen in the index
+	// Add all the child images described in the index
 	for _, manifest := range indexManifest.Manifests {
-		archImgDigest, err := fullDigest(descr.Ref.(name.Tag), manifest.Digest)
+		archImgDigest, err := fullDigest(tag, manifest.Digest)
 		if err != nil {
 			return nil, fmt.Errorf("generating digest for image: %w", err)
 		}

